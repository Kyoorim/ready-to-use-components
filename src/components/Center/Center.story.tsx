import React, { FC } from 'react';
import { Meta } from '@storybook/react';
import Center from './Center';
import { BoxProps } from 'components/Box';

export default {
  title: 'Center',
  component: Center,
} as Meta<BoxProps>;

export const Default: FC = () => {
  return <Center>Center</Center>;
};
