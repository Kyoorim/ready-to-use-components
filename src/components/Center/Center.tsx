import React from 'react';
import Box, { BoxProps } from '../Box/Box';
import { FCC } from 'types';

const CenterStyles = {
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
};

export const Center: FCC<BoxProps> = ({ children, ...props }) => {
  return (
    <Box {...CenterStyles} {...props} data-testid="center-comp">
      {children}
    </Box>
  );
};

export default Center;
