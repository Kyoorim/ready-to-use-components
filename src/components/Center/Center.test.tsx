import React from 'react';
import { render, screen } from '@testing-library/react';
import Center from './Center';

const defaultProps = {
  children: 'Center',
};

describe('<Center />', () => {
  it('should render', () => {
    render(<Center {...defaultProps} />);

    expect(screen.getByTestId('center-comp')).toBeInTheDocument();
  });

  it('should render children', () => {
    render(<Center {...defaultProps} />);

    expect(screen.getByText(/Center/i)).toBeInTheDocument();
  });

  it('should allow custom className', () => {
    const props = {
      ...defaultProps,
      className: 'Custom',
    };
    render(<Center {...props} />);

    expect(screen.getByTestId('center-comp')).toHaveProperty('className');
  });
});
