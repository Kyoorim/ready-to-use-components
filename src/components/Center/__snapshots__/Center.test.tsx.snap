// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Center /> should render 1`] = `
<div
  class="sc-bcXHqe kzKljj"
  data-testid="center-comp"
  display="flex"
>
  Center
</div>
`;
